Book-Your-Place/bookyourplace
# Refactor all public resources to only access bookings (rather than guests).

#### What's this PR do?
Refactor all public resources to only access bookings (rather than guests).
Deletes all public/guests/* controllers, views and specs.
Minor coding standards of touched files.

##### Background context

    Part of the work to allow a publicly accessible booking URL.

    For security reasons, the ONLY publicly accessible resources
    will be bookings, not guests.

    Unauthenticated users will be able to access a public/trips/:trip_id/bookings/new
    page, enter their card details and create a new booking on a trip.

#### Where should the reviewer start?
app/controllers/public/trips/bookings_controller.rb
 - this where the main logic of flow of user control is defined.

#### How should this be manually tested?
Get a trip id...
Open rails console:
`rails c`

`$ Trip.last.id`

Then visit:
/public/trips/:trip_id/bookings/new

Follow the self-explanatory UX...

Use Stripe test card data:

https://stripe.com/docs/testing#cards

Log into the Stripe dashboard, in the test environment and make sure you can see the money coming in!

#### Screenshots
￼
![image](https://user-images.githubusercontent.com/1453680/51441156-91661700-1cc6-11e9-909a-421e9a02a48c.png)

![image](https://user-images.githubusercontent.com/1453680/51441165-9e830600-1cc6-11e9-8804-f8bace12af2a.png)

![image](https://user-images.githubusercontent.com/1453680/51441171-a93d9b00-1cc6-11e9-9806-9bfc1a59da6f.png)



---

#### Migrations
none

#### Additional deployment instructions
none

#### Additional ENV Vars
none
